cmd/api/main.go

```diff
@@ -24,26 +27,46 @@ func loadConfig() (*Config, error) {
 
 	return &Config{
 		Addr: viper.GetString("ADDR"),
+		DatabaseDSN: fmt.Sprintf(
+			"postgres://%s:%s@%s:%s/%s",
+			viper.GetString("DATABASE_USER"),
+			viper.GetString("DATABASE_PASSWORD"),
+			viper.GetString("DATABASE_HOST"),
+			viper.GetString("DATABASE_PORT"),
+			viper.GetString("DATABASE_DB"),
+		),
 	}, nil
 }
 
 type ItemStore interface {
-	GetAll() hn.Items
-	GetStories() hn.Items
-	GetJobs() hn.Items
+	GetAll(ctx context.Context) ([]database.Item, error)
+	GetStories(ctx context.Context) ([]database.Item, error)
+	GetJobs(ctx context.Context) ([]database.Item, error)
 }
 
 func main() {
+	if err := run(); err != nil {
```

Comment:
I'd probably put your run in here, doesn't seem nessessary to extract it

Reply:
I move it out so all of the service setup error handling is done in one place. Do you think it'd be better to do the `log.Fatal` multiple times instead? e.g.

```
func main() {

 db, err := database.New()
 if err != nil {
    log.Fatal(err)
 }
 
  client, err := client.New()
  if err != nil {
    log.Fatal(err)
  }
  
  // etc, etc...
}
```

Reply:
Think thats fine as its the main, and its only setting up the server